src/main/java/com/mulberry/blackmulberry/service/UserService.java

```diff
@@ -0,0 +1,32 @@
+package com.mulberry.blackmulberry.service;
+
+import com.mulberry.blackmulberry.entity.User;
+import com.mulberry.blackmulberry.exception.DuplicateResourceException;
+import com.mulberry.blackmulberry.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    public void save(User user) throws DuplicateResourceException {
+        if (userRepository.findByEmail(user.getEmail()).isPresent()){
```

Comment:
its okay, but try using Optional's orElseThrow method here. and you can use lambda () ->